react-demo-app/src/components/Video.tsx

```diff
@@ -0,0 +1,184 @@
+      dispatch(forceVideoConversationUpdate({ conversationId: session.conversationId }));
+    });
+
+    session.on('participantsUpdate', partsUpdate => {
+      const participant = partsUpdate.activeParticipants.find(part => part.userId === session.fromUserId);
+      const conversation = videoConversations.find(c => c.conversationId === partsUpdate.conversationId);
+      const participantStore = conversation?.participantsUpdate?.activeParticipants?.find(p => p.userId === session.fromUserId);
+      if (participant?.audioMuted !== participantStore?.audioMuted) {
+        dispatch(updateAudioLoading({loading: false, convId: partsUpdate.conversationId}));
+      }
+      if (participant?.videoMuted !== participantStore?.videoMuted) {
+        dispatch(updateVideoLoading({loading: false, convId: partsUpdate.conversationId}));
+      }
+
+      dispatch(addParticipantUpdateToVideoConversation(partsUpdate));
+      setIsIncomingStreamActive(partsUpdate.activeParticipants.length < 2 ? false :
+        !partsUpdate.activeParticipants.find(part => session.fromUserId !== part.userId)?.videoMuted);
+      setIsOutgoingStreamActive(
+        !partsUpdate.activeParticipants.find(part => session.fromUserId === part.userId)?.videoMuted);
+    });
+
+    session.on('terminated', reason => {
+      dispatch(removeVideoConversationFromActive({conversationId: session.conversationId, reason: reason}));
+      setIsIncomingStreamActive(false);
+      setIsOutgoingStreamActive(false);
+    });
+  }
+
+  return (
+    <>
+      <Card className="video-card">
+        <h2>Video</h2>
+        <div className="buttons-and-video-container">
+          <input
+            value={roomJid}
+            onChange={(e) => setRoomJid(e.target.value)}
+            className="conference-input"
+          />
+          <div className="place-conference">
+            <GuxButton
+              accent="primary"
+              onClick={startVideoConf}
+            >
+              Join with roomJid
+            </GuxButton>
+            <GuxButton
+              accent="primary"
+              onClick={startVideoMeeting}
+            >
+              Join with conferenceId
+            </GuxButton>
+            <GuxButton
+              onClick={startScreenShare}
+            >
+              Screen Share
+            </GuxButton>
+          </div>
+          <div className="video-container-container">
+            <audio ref={audioRef} autoPlay/>
+            <div>
+              <p>Theirs</p>
+              <div className="video-container">
+                <video ref={videoRef} autoPlay playsInline
+                       style={{visibility: incomingStreamIsActive ? 'visible' : 'hidden'}}
+                />
+              </div>
+            </div>
+            <div>
+              <p>Yours</p>
+              <div className='video-container'>
+                <video ref={vanityVideoRef} autoPlay playsInline
+                       style={{visibility: outgoingStreamIsActive && !(
```

Comment:
Can some of this be put into a function instead? I don't love having a bunch of logic in the template if we can prevent it, I think its better defined above.